### ISSUE TITLE
Array "st" returned by "spikedetector3PC" has the wrong dimensions?

### ISSUE THREAD

**Issue by jamespherman** [MouseLand/Kilosort#475]
First time trying to use kilosort3. On line 83 of "extract_spikes", the function "spikedetector3PC" returns multiple arguments including "st" which (in my case) has dimensions: [4, 39]. On lines 97/98 it appears that "st" is being augmented with two extra rows by the commands: "st(5,:) = cF;" and "st(6,:) = k-1;", but MATLAB returns an error from this line. If I am running the main script, I get the error:

"Error using gpuArray/subsasgn
An unexpected error occurred trying to launch a kernel. The CUDA error was:
invalid configuration argument"

And if I stop execution (with "dbstop if error") and try to execute "st(5,:) = cF;", I get the error:

"Error using gpuArray/subsref
Index exceeds matrix dimension."

Have I done something wrong? Is it possible that something about how CUDA / MATLAB handles gpuArrays and augmenting their size has changed? Any info on debugging would be very helpful...

**Comment by LapishLab** [comment]
I have this same error. Any insight is much appreciated. 

Matlab 2021b, Quadro P4000, Ubuntu 20.04

**Comment by LapishLab** [comment]
It seems this solves the issue for me:

https://github.com/MouseLand/Kilosort/issues/427

**Comment by jamespherman** [comment]
> It seems this solves the issue for me:
> 
> #427

Thanks for the suggestion! I was able to work around that particular line of code, but it appears that I'm having more systemic problems with CUDA / gpuArray operations. For example, with that particular line if I use "dbstop if error", MATLAB stops there, but if I then execute the line again it executes just fine. There are many places in the Kilosort code where this happens. I've been in touch with Marius, and he's using the came card and matlab versions as me (Matlab 2021a / GeForce 3090), so I'm not certain what the problem is. Marius suggested I start by uninstalling visual studio, matlab, and cuda, and starting fresh which I'm in the process of doing right now.

I've also found that other MATLAB CUDA users have this same problem (error on first execution that then works the next time around), which could be related to how long windows allows a command to run before it times out. If a fresh install doesn't fix this for me, I'm going to open a ticket with the Mathworks and have them help me figure out what's going on.
